src/txn/txn_log.c

```diff
@@ -253,26 +253,62 @@ __wt_txn_log_op(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt)
     WT_DECL_RET;
     WT_ITEM *logrec;
     WT_TXN *txn;
-    WT_TXN_OP *op;
+    WT_TXN_OP *op, *prevop;
+    WT_UPDATE *upd;
+    bool resolve_uncommitted, undo_resolving_uncommitted;
 
     uint32_t fileid;
 
     conn = S2C(session);
     txn = session->txn;
+    undo_resolving_uncommitted = false;
 
     /* We'd better have a transaction. */
     WT_ASSERT(session, F_ISSET(txn, WT_TXN_RUNNING) && F_ISSET(txn, WT_TXN_HAS_ID));
 
     WT_ASSERT(session, txn->mod_count > 0);
     op = txn->mod + txn->mod_count - 1;
     fileid = op->btree->id;
+    resolve_uncommitted = (txn->resolve_uncommitted && !WT_IS_METADATA(op->btree->dhandle) &&
```

Comment:
It definitely feels a bit weird having a resolve_uncommitted that isn't equal to the other resolve_uncommitted. I think we should improve this section.

Reply:
Good point - I have updated it to "set_weak_hazard" , because the local variable is really to decide if we have to set a weak hazard or not. Also fixed the ugly loop to find the next update from the same txn. Resolving the comment.